File: springboot/src/main/java/com/freshmall/entity/Goods.java
package com.freshmall.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品表
 */
@Data
@Entity
@Table(name = "goods")
public class Goods {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "category_id", nullable = false)
    private Integer category_id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 255)
    private String image = "";

    /** 轮播图 JSON 数组 */
    @Column(columnDefinition = "TEXT")
    private String images;

    /** 图文详情富文本 */
    @Column(columnDefinition = "TEXT")
    private String detail;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(name = "original_price", precision = 10, scale = 2)
    private BigDecimal original_price;

    @Column(length = 20)
    private String unit = "份";

    private Integer stock = 0;

    private Integer sales = 0;

    @Column(name = "is_hot")
    private Boolean is_hot = false;

    @Column(name = "is_new")
    private Boolean is_new = false;

    /** 上架状态 */
    private Boolean status = true;

    @Column(name = "created_at", updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: server/models/goods.js
/**
 * 商品模型 → goods 表
 */
const { DataTypes } = require('sequelize')
const sequelize = require('../config/database')

const Goods = sequelize.define('Goods', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  category_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    comment: '所属分类 id',
  },
  name: { type: DataTypes.STRING(100), allowNull: false, comment: '商品名称' },
  image: { type: DataTypes.STRING(255), defaultValue: '', comment: '主图' },
  images: { type: DataTypes.TEXT, comment: '轮播图 JSON 数组' },
  detail: { type: DataTypes.TEXT, comment: '图文详情（富文本）' },
  price: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    comment: '现价（元）',
  },
  original_price: { type: DataTypes.DECIMAL(10, 2), comment: '划线价（元）' },
  unit: { type: DataTypes.STRING(20), defaultValue: '份', comment: '销售单位，如 500g/份' },
  stock: { type: DataTypes.INTEGER, defaultValue: 0, comment: '库存' },
  sales: { type: DataTypes.INTEGER, defaultValue: 0, comment: '销量' },
  is_hot: { type: DataTypes.BOOLEAN, defaultValue: false, comment: '是否热销' },
  is_new: { type: DataTypes.BOOLEAN, defaultValue: false, comment: '是否新品' },
  status: { type: DataTypes.BOOLEAN, defaultValue: true, comment: '上架状态，1 上架 0 下架' },
}, {
  tableName: 'goods',
})

module.exports = Goods
